public/js/utils.js

```diff
@@ -0,0 +1,93 @@
+async function enableEthereum() {
+  if (window.ethereum) {
+    metamask = new Web3(window.ethereum);
+    try {
+      await window.ethereum.enable();
+      console.log('ethereum enabled...');
+    } catch(error) {
+      console.error(error);
+    }
+  }
+  else if (window.web3) {
+    window.web3 = new Web3(window.web3.currentProvider);
+    console.log('Legacy web3 detected!');
+  }
+  else {
+    console.log('Non-Ethereum browser detected!');
+    return;
+  }
+}
+
+
+function buf2hex(buffer) { // buffer is an ArrayBuffer
+  return Array.prototype.map.call(new Uint8Array(buffer), x => ('00' + x.toString(16)).slice(-2)).join('');
+}
+
+function submitTransaction() {
+  var wastCode = document.getElementById('wastCode').value;
+  var value = document.getElementById('value').value;
+
+  var wast = "";
+  var wasm = "";
+  var bytes = 0;
+
+  // Validate wast
+  if (wastCode.length > 0) {
+    console.log(wastCode);
+    // Compile wast to wasm
+    try {
+      var module = window.Binaryen.parseText(wastCode);
```

Comment:
Why do you need to use binaryen? Is the wabt already supplied not enough?

Reply:
changed the code to use wabt.